src/SwedbankPay.Sdk.Infrastructure/JsonSerialization/TypesafeEnumConverter.cs

```diff
@@ -1,121 +1,121 @@
+
+            if (reader.TokenType == JsonToken.String)
+            {
+                var enumText = reader.Value.ToString();
+
+                var val = FromValue(objectType, enumText);
+
+                if (val != null)
+                    return val;
+            }
+            else if (reader.TokenType == JsonToken.Integer)
+            {
+                var enumVal = Convert.ToInt32(reader.Value);
+                var values = (int[])Enum.GetValues(objectType);
+                if (values.Contains(enumVal))
+                {
+                    return Enum.Parse(objectType, enumVal.ToString());
+                }
+            }
+
+            var names = Enum.GetNames(objectType);
+
+            var unknownName = names
+                .Where(n => string.Equals(n, UnknownValue, StringComparison.OrdinalIgnoreCase))
+                .FirstOrDefault();
+
+            if (unknownName == null)
+            {
+                throw new JsonSerializationException($"Unable to parse '{reader.Value}' to enum {objectType}. Consider adding Unknown as fail-back value.");
+            }
+
+            return Enum.Parse(objectType, unknownName);
+        }
+
+        public override void WriteJson(JsonWriter writer, object value, JsonSerializer serializer)
+        {
+            Type enumType = value.GetType();
+
+            InitMap(enumType);
+
+            var val = ToValue(enumType, value);
+
+            writer.WriteValue(val);
+        }
+
+        #region Private methods
+        private void InitMap(Type enumType)
+        {
+            if (this._fromValueMap == null)
+                this._fromValueMap = new Dictionary<Type, Dictionary<string, object>>();
+
+            if (this._toValueMap == null)
+                this._toValueMap = new Dictionary<Type, Dictionary<object, string>>();
+
+            if (this._fromValueMap.ContainsKey(enumType))
+                return; // already initialized
+
+            var fromMap = new Dictionary<string, object>(StringComparer.InvariantCultureIgnoreCase);
+            var toMap = new Dictionary<object, string>();
+
+            FieldInfo[] fields = enumType.GetFields(BindingFlags.Static | BindingFlags.Public);
+
+            foreach (FieldInfo field in fields)
+            {
+                var name = field.Name;
+                var enumValue = Enum.Parse(enumType, name);
+
+                toMap[enumValue] = name;
+                fromMap[name] = enumValue;
+            }
+
+            this._fromValueMap[enumType] = fromMap;
+            this._toValueMap[enumType] = toMap;
+        }
+
+        private string ToValue(Type enumType, object obj)
+        {
+            Dictionary<object, string> map = this._toValueMap[enumType];
+
+            return map[obj];
+        }
+
+        private object FromValue(Type enumType, string value)
+        {
+            Dictionary<string, object> map = this._fromValueMap[enumType];
+
+            if (!map.ContainsKey(value))

```

Comment:
Codacy found an issue: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-sdk-dotnet/pullRequest?prid=6290774)